hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/StakeChangeManager.java

```diff
@@ -0,0 +1,181 @@
+		return (entityId < 0) ? 0 : entityId;
+	}
+
+	public long getNodeStakeeNum(final Map<AccountProperty, Object> changes) {
+		final var entityId = (long) changes.getOrDefault(STAKED_ID, 0L);
+		// Node ids are negative
+		return (entityId < 0) ? entityId : 0;
+	}
+
+	public static long finalBalanceGiven(
+			@Nullable final MerkleAccount account,
+			@NotNull final Map<AccountProperty, Object> changes
+	) {
+		if (changes.containsKey(BALANCE)) {
+			return (long) changes.get(BALANCE);
+		} else {
+			return (account == null) ? 0 : account.getBalance();
+		}
+	}
+
+	public boolean finalDeclineRewardGiven(
+			@Nullable final MerkleAccount account,
+			@NotNull final Map<AccountProperty, Object> changes
+	) {
+		if (changes.containsKey(DECLINE_REWARD)) {
+			return (Boolean) changes.get(DECLINE_REWARD);
+		} else {
+			return account != null && account.isDeclinedReward();
+		}
+	}
+
+	public long finalStakedToMeGiven(
+			@Nullable final MerkleAccount account,
+			@NotNull final Map<AccountProperty, Object> changes
+	) {
+		if (changes.containsKey(STAKED_TO_ME)) {
+			return (long) changes.get(STAKED_TO_ME);
+		} else {
+			return (account == null) ? 0 : account.getStakedToMe();
+		}
+	}
+
+	public void updateStakedToMe(
+			@NotNull final int stakeeI,
+			final long delta,
+			@NotNull final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges
+	) {
+		final var mutableChanges = pendingChanges.changes(stakeeI);
+		if (mutableChanges.containsKey(STAKED_TO_ME)) {
+			mutableChanges.put(STAKED_TO_ME, (long) mutableChanges.get(STAKED_TO_ME) + delta);
+		} else {
+			mutableChanges.put(STAKED_TO_ME, delta);
+		}
+		pendingChanges.updateChange(stakeeI, mutableChanges);
+	}
+
+	public void updateBalance(
+			final long delta,
+			final int rewardAccountI,
+			@NotNull final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges
+	) {
+		final var mutableChanges = pendingChanges.changes(rewardAccountI);
+		if (mutableChanges.containsKey(BALANCE)) {
+			mutableChanges.put(BALANCE, (long) mutableChanges.get(BALANCE) + delta);
+		} else {
+			mutableChanges.put(BALANCE, delta);
+		}
+		pendingChanges.updateChange(rewardAccountI, mutableChanges);
+	}
+
+	public static boolean hasStakeFieldChanges(@NotNull final Map<AccountProperty, Object> changes) {
+		return changes.containsKey(BALANCE) ||
+				changes.containsKey(DECLINE_REWARD) ||
+				changes.containsKey(STAKED_ID) ||
+				changes.containsKey(STAKED_TO_ME);
+	}
+
+	public int findOrAdd(
+			final long accountNum,
+			final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges
+	) {
+		final var n = pendingChanges.size();
+		for (int i = 0; i < n; i++) {
+			if (pendingChanges.id(i).getAccountNum() == accountNum) {
+				return i;
+			}
+		}
+		// This account wasn't in the current change set
+		pendingChanges.include(
+				STATIC_PROPERTIES.scopedAccountWith(accountNum),
+				accounts.get().get(EntityNum.fromLong(accountNum)),
```

Comment:
⚠️ This must be `accouns.get().getForModify()` because of [the way `TransactionalLedger` uses](https://github.com/hashgraph/hedera-services/blob/9a04d22d5087a5e766ac3c0379c73040176ac5c8/hedera-node/src/main/java/com/hedera/services/ledger/TransactionalLedger.java#L530) these cached entities.